docs/docs/api_documentation_style.md

```diff
@@ -105,68 +125,73 @@ In order, the sections of a funciton comment are:
 	
 	Function handles have type `function_handle`.
 
-5. Returns
+2. Returns
 
-   Explanation of the returned values and their types. We distinguish two cases.
+    Explanation of the returned values and their types. We distinguish two cases.
    
-   First, when a single value is returned, we use the Google style:
+    First, when a single value is returned, we use the Google style:
    
-   ```matlab
-   function c = sum(a, b)
-   % Sums two numbers
-   %
-   % Returns:
-   %   double: The sum of the parameters
-   ```
+    ```matlab
+    function c = sum(a, b)
+    % Sums two numbers
+    %
+    % Returns:
+    %   double: The sum of the parameters
+    ```
    
    
-   ```matlab
-   function [c d] = sorted2(a, b)
-   % Sorts two numbers
-   %
-   % Returns
-   % -------
-   %   c: double
-   %     Smallest number
-   %   d: double
-   %     Largest number
-   ```
+    ```matlab
+    function [c d] = sorted2(a, b)
+    % Sorts two numbers
+    %
+    % Returns
+    % -------
+    %   c: double
+    %     Smallest number
+    %   d: double
+    %     Largest number
+    ```
    
-6. Raises
+3. Raises (optional)
 
 	An optional section detailling which errors get raised and under what conditions.
 	
 	TO BE COMPLETED
 	
-7. Warns
+4. Warns (optional)
 
 	An optional section detailling which warnings get raised and under what conditions, formatted similarly to Raises.
 	
-8. Warnings
+5. Warnings (optional)
 
 	An optional section with cautions to the user in free text/reST.
+
+6. Examples (encouraged)
+
+	A section with explicit commands illustrating as clearly as possible one or several ways of calling the function.
+
+	(TODO: allow examples that would be executed by the test framework, possibly with doctests, to be tested)
 	
-9. See Also
+7. See Also (encouraged)
 
-	An optional section used to refer to related code.
+	An optional section used to refer to related code. This will allow for easily browsing the documentation.
 	
 	TO BE COMPLETED
 	
-10. Notes
+8. Notes (optional)
+
+	This optional section can provide various additional information about the code of interest to the user, such as a discussion about the algorithm used by the function. Depreciation warnings can also be specified here. We haven't yet specified a formal syntax for those. The numpydoc convention uses a Sphinx directive which we tend to avoid.
+
+	Before the section, the comment should be broken by a single empty line without `%`. This should stop matlab from parsing, and thus allow us to use further formatting such as LaTeX. The content from now on would then only be presented in the Sphinx API. (TODO: check this)
```

Comment:
@denisrosset I suspect that breaking the comment section here, as in 

```
function tagada
% tagada - strawberry

% Notes:
%     tagada
```

would allow to define documentation that appear in sphinx but not matlab. Thus we could use latex and other fancy formattings there. Maybe we can check this when trying the sphinx stuff. We should only do this for the minimum set of options (`Notes` and `References`), since anything below this line break would not appear on the repl!

Reply:
I just tested, Sphinx would not show the Notes section in that case. So we can't allow a line break, or we'd need to hack the Sphinx Matlab domain.

Reply:
That's strange, I used to see something like that working... ok we'll need to explore this further!